ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/backwardsync/ForwardSyncStep.java

```diff
@@ -68,6 +80,29 @@ protected BlockHeader processKnownAncestors(final Void unused) {
     return null;
   }
 
+  @VisibleForTesting
+  protected List<BlockHeader> returnFirstNUnknownHeaders(final Void unused) {
+    while (backwardChain.getFirstAncestorHeader().isPresent()) {
+      BlockHeader header = backwardChain.getFirstAncestorHeader().orElseThrow();
+      if (context.getProtocolContext().getBlockchain().contains(header.getHash())) {
+        debugLambda(
+            LOG,
+            "Block {} is already imported, we can ignore it for the sync process",
+            () -> header.getHash().toString().substring(0, 20));
+        backwardChain.dropFirstHeader();
+      } else if (backwardChain.isTrusted(header.getHash())) {
+        debugLambda(
+            LOG,
+            "Block {} was added by consensus layer, we can trust it and should therefore import it.",
```

Comment:
"Block {} was added by trusted consensus layer, importing it.",

Reply:
Importing trusted block {}